src/main/java/com/powsybl/openloadflow/network/LfZeroImpedanceNetwork.java

```diff
@@ -0,0 +1,215 @@
+                    subGraph.addVertex(bus2);
+                }
+                if (!branch.isDisabled()) {
+                    subGraph.addEdge(bus1, bus2, branch);
+                }
+            }
+        }
+        return subGraph;
+    }
+
+    public LfNetwork getNetwork() {
+        return network;
+    }
+
+    public boolean isDc() {
+        return dc;
+    }
+
+    public Graph<LfBus, LfBranch> getGraph() {
+        return graph;
+    }
+
+    public SpanningTreeAlgorithm.SpanningTree<LfBranch> getSpanningTree() {
+        return spanningTree;
+    }
+
+    public void updateSpanningTree() {
+        spanningTree = new KruskalMinimumSpanningTree<>(graph).getSpanningTree();
+        Set<LfBranch> spanningTreeEdges = spanningTree.getEdges();
+        for (LfBranch branch : graph.edgeSet()) {
+            branch.setSpanningTreeEdge(dc, spanningTreeEdges.contains(branch));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void linkVoltageControls(VoltageControl<?> mainVc, VoltageControl<?> vc) {
+        mainVc.mergedDependentVoltageControls.add((VoltageControl) vc);
+        vc.mainMergedVoltageControl = (VoltageControl) mainVc;
+    }
+
+    public void updateVoltageControlMergeStatus() {
+        Map<VoltageControl.Type, List<VoltageControl<?>>> voltageControlsByType = new EnumMap<>(VoltageControl.Type.class);
+        for (LfBus zb : graph.vertexSet()) { // all enabled by design
+            for (VoltageControl<?> vc : zb.getVoltageControls()) {
+                voltageControlsByType.computeIfAbsent(vc.getType(), k -> new ArrayList<>())
+                        .add(vc);
+                vc.getMergedDependentVoltageControls().clear();
+                vc.mainMergedVoltageControl = null;
+                vc.mergeStatus = VoltageControl.MergeStatus.MAIN;
+            }
+        }
+        for (List<VoltageControl<?>> voltageControls : voltageControlsByType.values()) {
+            if (voltageControls.size() > 1) {
+                // we take the highest target voltage (why not...) and in case of equality the voltage control
+                // with the first controlled bus ID by alpha sort
+                voltageControls.sort(Comparator.<VoltageControl<?>>comparingDouble(VoltageControl::getTargetValue)
+                        .reversed()
+                        .thenComparing(o -> o.getControlledBus().getId()));
+                VoltageControl<?> mainVc = voltageControls.get(0);
+                // first one is main, the other have are dependents
+                for (int i = 1; i < voltageControls.size(); i++) {
+                    VoltageControl<?> vc = voltageControls.get(i);
+                    vc.mergeStatus = VoltageControl.MergeStatus.DEPENDENT;
+                    linkVoltageControls(mainVc, vc);
+                }
+            }
+        }
+    }
+
+    public void tryToSplit() {
+        List<LfBranch> disabledBranches = graph.edgeSet().stream()
```

Comment:
@geofjamg and @flo-dup here we remove edges linked to disabled branches. I am not sure to understand where we do the reverse action and if it is needed ?